components/packages/pb-input/src/pb-input.ts

```diff
@@ -0,0 +1,190 @@
+  //
+  // @property({ attribute: 'hidden', type: Boolean, reflect: true })
+  // private hidden: boolean = false;
+
+  @property({ attribute: 'required', type: Boolean, reflect: true })
+  private required: boolean = false;
+
+  @property({ attribute: 'min-length', type: String, reflect: true })
+  private minLength: number = 0;
+
+  @property({ attribute: 'max-length', type: String, reflect: true })
+  private maxLength: number = Number.MAX_VALUE;
+
+  @property({ attribute: 'readonly', type: Boolean, reflect: true })
+  private readOnly: boolean = false;
+
+  @property({ attribute: 'label-hidden', type: Boolean, reflect: true })
+  private labelHidden: boolean = false;
+
+  @property({ attribute: 'label', type: String, reflect: true })
+  private label: string = "Default label";
+
+  @property({ attribute: 'label-position', type: String, reflect: true })
+  private labelPosition: string = "top";
+
+  @property({ attribute: 'label-width', type: String, reflect: true })
+  private labelWidth: number = 4;
+
+  @property({ attribute: 'placeholder', type: String, reflect: true })
+  private placeholder: string = "";
+
+  @property({ attribute: 'value', type: String, reflect: true })
+  private value: string = "";
+
+  @property({ attribute: 'type', type: String, reflect: true })
+  private type: string = "text";
+
+  @property({ attribute: 'min', type: String, reflect: true })
+  private min: string = "";
+
+  @property({ attribute: 'max', type: String, reflect: true })
+  private max: string = "";
+
+  @property({ attribute: 'step', type: String, reflect: true })
+  private step: string = "1";
+
+
+  constructor() {
+    super();
+    listenForLangChanged(() => {
+      // TODO: enable translation
+      // We do not use translate() directive in render() since placeholder is an exposed attribute
+      // this.placeholder = get("placeholder");
+    });
+  }
+
+  async attributeChangedCallback(name: string, old: string|null, value: string|null) {
+    super.attributeChangedCallback(name, old, value);
+    if (name === 'lang') {
+      use(this.lang).then();
+    }
+  }
+
+  static getCatalog(lang: string) {
+    switch(lang) {
+      case "es":
+      case "es-ES":
+        return i18n_es;
+      case "fr":
+        return i18n_fr;
+      case "ja":
+        return i18n_ja;
+      case "pt":
+      case "pt-BR":
+        return i18n_pt;
+      default:
+        return i18n_en;
+    }
+  }
+
+  static get styles() {
+    return css`
+      :host {
+        display: block;
+        font-family: sans-serif;
+        text-align: left;
+        padding: 10px 10px 10px 0px;
+        max-width: var(--max-width, 50%);
+      }
+
+      .input-elem {
+        font-size: 14px;
+        height: 20px;
+      }
+      
+      .label-elem {
+        font-size: 14px;
+        font-weight: 700;
+        margin-bottom: 5px;
+      }
+      
+      /* Add a red star after required inputs */
+      .label-required:after {
+        content: "*";
```

Comment:
put space before * to be more confortable display

Reply:
yes, agree